### PR TITLE
[JENKINS-61409] [Targeted for 2.222.x] Upgrade to 4.2.1 version of Remoting.

### PR DESCRIPTION
**Pull request by jeffret-b**
This is an interim Remoting release intended for the 2.222.x LTS release. It contains only the WebSocket fixes for JENKINS-61409 added onto the 4.2 release. There are no other changes beyond what is already contained in Remoting 4.2 for the 2.222 release.

This version of Remoting provides a fix to the WebSocket implementation for large payloads. See [JENKINS-61409](https://issues.jenkins-ci.org/browse/JENKINS-61409). This Remoting upgrade needs to be paired with #4596 to complete the master side of the correction. This change is breaking in terms of WebSockets but as this feature is still considered Beta, this is not considered a breaking change for the agent or server generally.

See the [Remoting changelog](https://github.com/jenkinsci/remoting/releases/tag/remoting-4.2.1)
### Proposed changelog entries

* Upgrade to Remoting 4.2.1 to fix an issue with large payloads over WebSockets. Requires an agent upgrade for the Beta WebSocket feature. (issue 61409)
### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
Remoting library update. Tests have been added in the related PRs.
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Comment by jglick** [comment]
This should only be paired with the changes from #4596.

**Review by oleg-nenashev — DISMISSED** [review]
It is incomplete. https://github.com/jenkinsci/jenkins/pull/4596 is also needed to fix the JENKINS-61409 defect in the LTS baseline

**Comment by jeffret-b** [comment]
Should I merge #4596 into this one as a complete PR? I wasn't sure it if be more confusing to do that or less.

File: test/src/test/java/jenkins/agents/WebSocketAgentsTest.java [hunk 1]
```diff
@@ -99,9 +100,10 @@ public void smokes() throws Exception {
             ).stdout(System.out).start());
             r.waitOnline(s);
             assertEquals("response", s.getChannel().call(new DummyTask()));
+            assertNotNull(s.getChannel().call(new FatTask()));
             FreeStyleProject p = r.createFreeStyleProject();
             p.setAssignedNode(s);
-            p.getBuildersList().add(Functions.isWindows() ? new BatchFile("echo hello") : new Shell("echo hello"));
+            p.getBuildersList().add(Functions.isWindows() ? new BatchFile("echo hello") : new Shell("dd if=/dev/random count=1024 bs=200"));
```
Comment (jglick) [inline test/src/test/java/jenkins/agents/WebSocketAgentsTest.java:106]: Suggest also cherry-picking #4605; i.e., reverting this line.

**Review by jglick — APPROVED** [review]

File: test/src/test/java/jenkins/agents/WebSocketAgentsTest.java (hunk 1 above)
Comment (jeffret-b) [reply test/src/test/java/jenkins/agents/WebSocketAgentsTest.java]: Thanks. I was thinking there was another change I was missing, but I wasn't finding it.

**Comment by jglick** [comment]
Not currently proposed for 2.222.2: https://github.com/jenkinsci/jenkins/pull/4618#issuecomment-610312047

**Comment by oleg-nenashev** [comment]
> Not currently proposed for 2.222.2

I raised it in Jenkins Jira FWIW

**Review by oleg-nenashev — APPROVED** [review]
Looks right. thanks for also backporting the test fix @jglick !

**Comment by olivergondza** [comment]
@jeffret-b, thanks for the work you have done to make this happen, but I still prefer to not to include this into 2.222.2. The change is still not soaked enough and while dedicated release minimizes negative impact of unrelated changes, it also introduce a version that received no soaking at all.

One thing that would make PRs like this more likely to be merged in the future, is to make strictly all commits `cherry-pick`ed with `-x` option to clearly show where the commit is coming from. I am of course happy to prepare the backports myself based on standard backporting process.

**Review by alecharp — APPROVED** [review]

**Comment by alecharp** [comment]
I'm adding the `ready-for-merge` label on this PR. Unless the PR receives any negative feedback, it should be merge in the next 24hrs. Thank you @jeffret-b.

**Comment by olivergondza** [comment]
@alecharp, careful. This is PR against a stable branch, not master. Merging it would mean a backport _after_ the RC is created, so it would be shipped in LTS without testing. Let me file a pro forma review to clearly indicate this cannot get merged just like that.

**Review by olivergondza — CHANGES_REQUESTED** [review]
Formally requesting changes so this does not get merged until next backporting window when it will be reconsidered again.

**Comment by jeffret-b** [comment]
We can convert this to draft or something if it would help track it more meaningfully until the next time.

**Comment by daniel-beck** [comment]
The only tracking needed is in the labels of the Jira issue IMO. I would close this, but wonder why @olivergondza kept it open.

**Comment by olivergondza** [comment]
This has been incorporated into https://github.com/jenkinsci/jenkins/pull/4710. Thanks!

**Pull request closed without merging**